pkg/codegen/go/gen_program_expressions.go

```diff
@@ -229,6 +229,9 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 		g.genNYI(w, "ReadFile")
 	case "readDir":
 		contract.Failf("unlowered toJSON function expression @ %v", expr.SyntaxNode().Range())
+	case "secret":
+		// TODO: generate an appropriate type assertion
```

Comment:
I'll follow an issue to track fixing this.

Reply:
Should be a pretty simple change if you apply that patch I mentioned earlier:

```diff
diff --git a/pkg/codegen/go/gen_program_expressions.go b/pkg/codegen/go/gen_program_expressions.go
index a3a61c9b..486d6fe4 100644
--- a/pkg/codegen/go/gen_program_expressions.go
+++ b/pkg/codegen/go/gen_program_expressions.go
@@ -622,7 +622,7 @@ func (g *generator) argumentTypeName(expr model.Expression, destType model.Type,
 				return "pulumi.Float64"
 			}
 			return "float64"
-		case model.StringType, model.DynamicType:
+		case model.StringType:
 			if isInput {
 				return "pulumi.String"
 			}
@@ -632,6 +632,11 @@ func (g *generator) argumentTypeName(expr model.Expression, destType model.Type,
 				return "pulumi.Bool"
 			}
 			return "bool"
+		case model.DynamicType:
+			if isInput {
+				return "pulumi.Any"
+			}
+			return "interface{}"
 		default:
 			return destType.Name
 		}

```

and then call `argumentTypeName` here.

Reply:
Done; thanks!